SPECS/xapi-storage-script.spec

```diff
@@ -19,55 +18,57 @@ BuildRequires: message-switch-devel
 BuildRequires: ocaml-rpc-devel
 BuildRequires: xapi-storage-ocaml-plugin-devel
 BuildRequires: ocaml-xcp-rrd-devel
+BuildRequires: systemd-devel
+
+Requires(post):   systemd
+Requires(preun):  systemd
+Requires(postun): systemd
 
 %description
 Allows script-based Xapi storage adapters.
 
 %prep 
 %setup -q -n %{name}-%{version}
-cp %{SOURCE1} xapi-storage-script-init
-cp %{SOURCE2} xapi-storage-script-conf.in
 
 %build
 make
 mv main.native xapi-storage-script
-./xapi-storage-script --help=groff > xapi-storage-script.1
-sed -e "s|@LIBEXECDIR@|%{_libexecdir}|g" xapi-storage-script-conf.in > xapi-storage-script.conf
+./xapi-storage-script --help=groff > xapi-storage-script.1 && gzip xapi-storage-script.1
+sed -e "s|@LIBEXECDIR@|%{_libexecdir}|g" %{SOURCE3} > xapi-storage-script.conf
 
 %install
- 
-mkdir -p %{buildroot}/%{_sbindir}
-install -m 0755 xapi-storage-script %{buildroot}/%{_sbindir}/xapi-storage-script
-mkdir -p %{buildroot}%{_sysconfdir}/init.d
-install -m 0755 xapi-storage-script-init %{buildroot}%{_sysconfdir}/init.d/xapi-storage-script
-mkdir -p %{buildroot}/%{_libexecdir}/xapi-storage-script
-mkdir -p %{buildroot}/%{_libexecdir}/xapi-storage-script/volume
-mkdir -p %{buildroot}/%{_libexecdir}/xapi-storage-script/datapath
-mkdir -p %{buildroot}/etc
-install -m 0644 xapi-storage-script.conf %{buildroot}/etc/xapi-storage-script.conf
-mkdir -p %{buildroot}%{_mandir}/man2
-install -m 0644 xapi-storage-script.1 %{buildroot}%{_mandir}/man2/xapi-storage-script.1
-gzip %{buildroot}%{_mandir}/man2/xapi-storage-script.1
+mkdir -p %{buildroot}%{_libexecdir}/xapi-storage-script
+mkdir -p %{buildroot}%{_libexecdir}/xapi-storage-script/volume
+mkdir -p %{buildroot}%{_libexecdir}/xapi-storage-script/datapath
+%{__install} -D -m 0755 xapi-storage-script %{buildroot}%{_sbindir}/xapi-storage-script
+%{__install} -D -m 0644 xapi-storage-script.conf %{buildroot}%{_sysconfdir}/xapi-storage-script.conf
+%{__install} -D -m 0644 xapi-storage-script.1.gz %{buildroot}%{_mandir}/man2/xapi-storage-script.1.gz
```

Comment:
The man page either needs to go in /usr/share/man8, and be renamed to xapi-storage-script.8.gz .